frontend/components/ToolCard.tsx

```diff
@@ -2,35 +2,60 @@ import React, { useState } from 'react';
 import ExpandLessIcon from '@mui/icons-material/ExpandLess';
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import {
-  Button, Card, CardActions, CardContent, CardMedia, Collapse, Grid, IconButton, Typography
+  Card, CardActions, CardContent, CardMedia, Collapse, Grid, IconButton, Typography
 } from '@mui/material';
 import { styled } from '@mui/material/styles';
 
 import AddRemoveButton from './AddRemoveButton';
 import DataElement from './DataElement';
+import { Tool } from '../interfaces';
 
 const ContainedCardMedia = styled(CardMedia)(({ theme }) => ({
   objectFit: 'contain',
   marginBottom: theme.spacing(2)
 }));
 
-export default function ToolCard(props) {
+interface CardImageProps {
+  image?: string
+  component: 'img'
+  height: number
+  alt: string
+}
+
+interface ToolCardProps {
+  tool: Tool
+}
+
+export default function ToolCard(props: ToolCardProps) {
   const { tool } = props;
 
   const [learnMoreActive, setLearnMoreActive] = useState(false);
 
+  let logoImageProps: CardImageProps = {
+    component: 'img',
+    height: 150,
+    alt: tool.logo_image_alt_text ?? 'Logo for tool'
+  };
+  if (tool.logo_image !== null) {
+    logoImageProps = { image: tool.logo_image, ...logoImageProps };
+  }
+
+  let mainImageProps: CardImageProps = {
+    component: 'img',
+    height: 150,
+    alt: tool.main_image_alt_text ?? 'Image of tool in use'
+  };
+  if (tool.main_image !== null) {
+    mainImageProps = { image: tool.main_image, ...mainImageProps };
+  }
```

Comment:
I had to do some strange stuff here to placate the conditional interface on `CardMedia` and also handle the `null` cases. I may return to to this later, especially if we add a default image, but this should be functional for now.